client/post-editor/editor-ground-control/index.jsx

```diff
@@ -294,29 +292,32 @@ export class EditorGroundControl extends PureComponent {
 						/>
 						{ this.getVerificationNoticeLabel() }{' '}
 						<span className="editor-ground-control__email-verification-notice-more">
-							{ this.props.translate( 'Learn More' ) }
+							{ translate( 'Learn More' ) }
 						</span>
 					</div>
 				) }
-				<div className="editor-ground-control__status">
-					{ this.isSaveEnabled() && (
-						<button
-							className="editor-ground-control__save button is-link"
-							onClick={ this.onSaveButtonClick }
-							tabIndex={ 3 }
-						>
-							{ this.props.translate( 'Save' ) }
-						</button>
-					) }
-					{ ! this.isSaveEnabled() && (
-						<span
-							className="editor-ground-control__save-status"
-							data-e2e-status={ this.getSaveStatusLabel( identity ) }
-						>
-							{ this.getSaveStatusLabel( this.props.translate ) }
-						</span>
-					) }
-				</div>
+				{ ( isSaveEnabled || shouldShowStatusLabel ) && (
+					<div className="editor-ground-control__status">
+						{ isSaveEnabled && (
+							<button
+								className="editor-ground-control__save button is-link"
+								onClick={ this.onSaveButtonClick }
+								tabIndex={ 3 }
+							>
+								{ translate( 'Save' ) }
+							</button>
+						) }
+						{ ! isSaveEnabled &&
+						shouldShowStatusLabel && (
+							<span
+								className="editor-ground-control__save-status"
+								data-e2e-status={ isSaving ? 'Saving…' : 'Saved' }
```

Comment:
I've opted for some slight repetition here in a trade-off between that and complexity. 
When I came to view this area of the code I found `this.getSaveStatusLabel` to be a little convoluted considering it's seemingly simple end goal.
Please just take some extra to consider the logical flow here and sanity check my approach - my aim was to see the same end result.